### PR TITLE
Generate specific platforms for NotImplemented types

### LINKED ISSUE #4893: Generated NotImplementedAttribute does not contain specific platform on types
## Current behavior

The `SymbolIcon` type is marked as not implemented, even if members are implemented.

## Expected behavior

The `SymbolIcon` is not marked as not implemented.

## Anything else we need to know?

The API generator should determine if inner elements have been implemented and include the proper not implemented list, like the `#if` statement above.

## Note for contributors

The code locations to update are referencing this block: 
https://github.com/unoplatform/uno/blob/93818bee2424961c31237c5c154ee35648cf9208/src/Uno.UWPSyncGenerator/Generator.cs#L277-L290

### PR DESCRIPTION
**Pull request by Youssef1313**
GitHub Issue (If applicable): closes #4893

## PR Type

What kind of change does this PR introduce?

- Bugfix

## What is the current behavior?

NotImplemented attribute doesn't have platforms when applied to a type.

## What is the new behavior?

Correctly generate platforms.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

**Comment by Youssef1313** [comment]
@jeromelaban This is ready for review.

**Comment by jeromelaban** [comment]
It looks like there are breaking changes introduced by this PR: https://github.com/unoplatform/uno/pull/6866/checks?check_run_id=3446252603

**Comment by Youssef1313** [comment]
@jeromelaban Fixed.

File: src/Uno.UI/Generated/3.0.0.0/Windows.UI.Xaml.Controls/TextBlock.cs [hunk 1]
```diff
@@ -371,13 +371,7 @@ public  void Select( global::Windows.UI.Xaml.Documents.TextPointer start,  globa
...
 		#endif
-		#if __ANDROID__ || __IOS__ || NET461 || __WASM__ || __SKIA__ || __NETSTD_REFERENCE__ || __MACOS__
-		[global::Uno.NotImplemented("__ANDROID__", "__IOS__", "NET461", "__WASM__", "__SKIA__", "__NETSTD_REFERENCE__", "__MACOS__")]
-		public  bool Focus( global::Windows.UI.Xaml.FocusState value)
-		{
-			throw new global::System.NotImplementedException("The member bool TextBlock.Focus(FocusState value) is not implemented in Uno.");
-		}
-		#endif
+		// Skipping already declared method Windows.UI.Xaml.Controls.TextBlock.Focus(Windows.UI.Xaml.FocusState)
 		// Forced skipping of method Windows.UI.Xaml.Controls.TextBlock.SelectionHighlightColor.get
```
Comment (Youssef1313) [inline src/Uno.UI/Generated/3.0.0.0/Windows.UI.Xaml.Controls/TextBlock.cs:375]: @jeromelaban Here, the package diff tool is reporting removed API for Focus method. The method is technically still there (inherited from UIElement).

The root bug here is that UIElement in UWP doesn't have Focus method, while it does in Uno. What's the action to take here?

Comment (Youssef1313) [reply src/Uno.UI/Generated/3.0.0.0/Windows.UI.Xaml.Controls/TextBlock.cs:375]: @jeromelaban Can you help here please?

Comment (jeromelaban) [reply src/Uno.UI/Generated/3.0.0.0/Windows.UI.Xaml.Controls/TextBlock.cs:375]: We're going to be adjust this in the api alignment branch as well.

Comment (Youssef1313) [reply src/Uno.UI/Generated/3.0.0.0/Windows.UI.Xaml.Controls/TextBlock.cs:375]: @jeromelaban So how to move forward with this PR?

Comment (jeromelaban) [reply src/Uno.UI/Generated/3.0.0.0/Windows.UI.Xaml.Controls/TextBlock.cs:375]: We cannot merge this PR in the main branch, so we can only fix this on the API alignments one.

Comment (Youssef1313) [reply src/Uno.UI/Generated/3.0.0.0/Windows.UI.Xaml.Controls/TextBlock.cs:375]: @jeromelaban Is that really necessary? I mean, is the PackageDiff reported error is actually a breaking change?

The simplified case is like the following:

```diff
public class BaseClass
{
    public void M() { }
}

public class DerivedClass : BaseClass
{
-    public void M() { }
}
```

It's like the PackageDiff reporting a removed API here. However, applications will still compile. The runtime behavior may change, but it should be for the better behavior.

Comment (Youssef1313) [reply src/Uno.UI/Generated/3.0.0.0/Windows.UI.Xaml.Controls/TextBlock.cs:375]: Ping @jeromelaban

Comment (Youssef1313) [reply src/Uno.UI/Generated/3.0.0.0/Windows.UI.Xaml.Controls/TextBlock.cs:375]: @jeromelaban Can you take a quick look please?

**Comment by jeromelaban** [comment]
Thanks for your contribution!

**Comment by Youssef1313** [comment]
@jeromelaban This was merged into api-alignments. I suppose the branch is no longer active and this should have re-targeted master?

**Pull request merged**